### ISSUE TITLE
Use Req.Request.halt/1 and return redirect exception instead of raising on "too many redirects"

### ISSUE THREAD

**Issue by bfolkens** [wojtekmach/req#320]
Perhaps there's a reason I'm not aware of for using raise during the "too many redirects" case, but I'm finding that this is more difficult to handle in a real-world scenario than if we were to halt the pipeline and return the exception instead.

For example, we could change line 1980:

https://github.com/wojtekmach/req/blob/67344bc9b5db0e241b42752f1d86f9ec594e96f8/lib/req/steps.ex#L1980

To the following:

```Elixir
{Req.Request.halt(request), RuntimeError.exception("too many redirects (#{max_redirects})")}
```

However, for backward compatibility we might want to include a `raise_on_redirect_threshold` or something similar.  If that's of interest I'm happy to work up a PR.

**Comment by wojtekmach** [comment]
Good call. I'm fine without `:raise_on_redirect_threshold`, I don't expect a lot of breakage from this particular change. Let's add a proper exception for this, say, `%Req.TooManyRedirects{max_redirects: n}`? A PR is very welcome!

**Comment by wojtekmach** [comment]
Fixed in #321.
